pkg/apis/v1alpha1/vpcnetworkconfiguration_types.go

```diff
@@ -0,0 +1,115 @@
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// SubnetType describes the subnet type.
+type SubnetType string
+
+const (
+	SubnetTypePublic  SubnetType = "public"
+	SubnetTypePrivate SubnetType = "private"
+)
+
+// VPCNetworkConfigurationSpec defines the desired state of VPCNetworkConfiguration
+type VPCNetworkConfigurationSpec struct {
+	// Namespaces to be applied
+	AppliedToNamespaces []string `json:"appliedToNamespaces,omitempty"`
+	// Tier0 gateway or Tier0 VRF path
+	// +kubebuilder:validation:Required
+	Tier0Gateway string `json:"tier0Gateway"`
+	// Edge cluster paths on which the networking elements will be created
+	// +kubebuilder:validation:Required
+	EdgeClusters []string `json:"edgeClusters"`
+	NSXTProject  string   `json:"nsxtProject,omitempty"`
+	// +kubebuilder:validation:MinItems=0
+	// +kubebuilder:validation:MaxItems=5
+	PublicIPV4Pools []string `json:"publicIPV4Pools,omitempty"`
+	// +kubebuilder:validation:MinItems=0
+	// +kubebuilder:validation:MaxItems=5
+	PrivateCIDRs []string `json:"privateCIDRs,omitempty"`
+	// +kubebuilder:validation:MinItems=1
+	// +kubebuilder:validation:MaxItems=1
+	IngressCIDRs       []string `json:"ingressCIDRs,omitempty"`
+	SubnetPrefixLength int      `json:"subnetPrefixLength,omitempty"`
+	// DefaultSubnetType defines the type of the default SubnetSet for PodVM and VM.
+	// Must be public or private.
+	// Defaults to public.
+	// +kubebuilder:validation:Enum=public;private
+	DefaultSubnetType SubnetType `json:"defaultSubnetType,omitempty"`
+}
+
+// VPCNetworkConfigurationStatus defines the observed state of VPCNetworkConfiguration
+type VPCNetworkConfigurationStatus struct {
+	// Conditions describes current state of VPCNetworkConfiguration.
+	Conditions []VPCNetworkConfigurationCondition `json:"conditions"`
+}
+
+// VPCNetworkConfigurationCondition defines the condition of VPCNetworkConfiguration.
+type VPCNetworkConfigurationCondition struct {
+	// Type defines the type of condition.
+	Type VPCNetworkConfigurationStatusCondition `json:"type"`
+	// Status shows the status of condition, one of True or False.
+	Status corev1.ConditionStatus `json:"status"`
+	// Reason shows a brief reason of condition.
+	Reason string `json:"reason,omitempty"`
+	// Message shows a human readable message about the condition.
+	Message string `json:"message,omitempty"`
+}
+
+// VPCNetworkConfigurationStatusCondition is an aspect of operator state.
+type VPCNetworkConfigurationStatusCondition string
+
+const (
+	// Ready indicates whether NSX Operator was able to successfully make
+	// a PATCH call with the user created VPCNetworkConfiguration CR
```

Comment:
Question - what does "PATCH call" mean? In which case, a VPCNetworkConfiguration is not ready?

Reply:
I will change the comment, the comment was copied from SecurityPolicy CR. Sorry for the confusion.
We will add some validation check for the fields in CR, if validation failed, will set the CR status to not ready,
for example, invalid CIDR set for privateCIDRs.